desc/plotting.py

```diff
@@ -3843,3 +3845,144 @@ def plot_logo(save_path=None, **kwargs):
         fig.savefig(save_path, facecolor=fig.get_facecolor(), edgecolor="none")
 
     return fig, ax
+
+
+def plot_adiabatic_invariant(
+    eq, rhos=None, alphas=None, num_pitch=None, pitch_idx=-1, mode="single-surface"
+):
+    """Plot the second adiabatic invariant J_|| (parallel action).
+
+    This function visualizes the parallel action invariant J_|| in either of two modes:
+    1. On a single flux surface as a function of fieldline label α and inverse
+       pitch angle 1/λ
+    2. As contours of J_|| for a fixed pitch angle across multiple flux surfaces
+
+    Parameters
+    ----------
+    eq : object
+        Equilibrium object containing magnetic field information
+    rhos : array_like or float, optional
+        Flux surface radii. If float, plots single surface.
+        Default: np.linspace(0.1, 1, 10)
+    alphas : array_like, optional
+        Fieldline label values (toroidal angle).
+        Default: np.linspace(0, 2π, 32, endpoint=True)
+    num_pitch : int, optional
+        Number of pitch angle values for bounce integral calculation.
+        Default: 16
+    pitch_idx : int, optional
+        Index of pitch angle to use for cross-section mode.
+        Default: -1 (last pitch angle)
+    mode : str, optional
+        Visualization mode, either "single-surface" or "cross-section".
+        Default: "single-surface"
+
+    Returns
+    -------
+    fig : matplotlib.figure.Figure
+        The figure object containing the plot
+    ax : matplotlib.axes.Axes
+        The axes object for further customization
+
+    Notes
+    -----
+    The second adiabatic invariant J_|| is calculated by integrating the parallel
```

Comment:
may also want to note our convention for handling multiple wells is to sum them

Reply:
Good point.